I have run the integration test for this PR in `eu-west-1` and seeing some errors, hope this help.
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ITestS3ARequesterPays.testRequesterPaysOptionSuccess:78 [Counter named stream_read_opened] 
Expecting:
 <0L>
to be greater than:
 <1L> 
[ERROR]   ITestS3AUnbuffer.testUnbuffer:75->Assert.assertTrue:53->Assert.assertTrue:42->Assert.fail:87
[ERROR]   ITestS3AUnbuffer.testUnbufferStreamStatistics:149->Assert.assertEquals:647->Assert.failNotEquals:835->Assert.fail:89 Count of stream_read_bytes starting=0 current=0 diff=0: stream_read_bytes expected:<8> but was:<0>
[ERROR]   ITestMarkerTool.testRunLimitedLandsatAudit:320->AbstractMarkerToolTest.runToFailure:271 Expected an exception of type class org.apache.hadoop.util.ExitUtil$ExitException
[ERROR] Errors: 
[ERROR]   ITestS3AContractUnbuffer>AbstractContractUnbufferTest.testUnbufferOnClosedFile:90->AbstractContractUnbufferTest.unbuffer:124 » IO
[ERROR]   ITestS3AContractStreamIOStatistics>AbstractContractStreamIOStatisticsTest.testInputStreamStatisticRead:253 » EOF
[INFO] 
[ERROR] Tests run: 1069, Failures: 4, Errors: 2, Skipped: 185
```
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   ITestS3AFileContextStatistics>FCStatisticsBaseTest.testStatistics:107 » EOF At...
[INFO] 
[ERROR] Tests run: 108, Failures: 0, Errors: 1, Skipped: 68
```

--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/common/ExecutorServiceFuturePool.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/common/ExecutorServiceFuturePool.java
@@ -61,7 +61,7 @@ public class ExecutorServiceFuturePool {
    */
   @SuppressWarnings("unchecked")
   public Future<Void> executeRunnable(final Runnable r) {
-    return (Future<Void>) executor.submit(() -> r.run());
+    return (Future<Void>) executor.submit(r::run);
   }
 
   public String toString() {

--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/common/TestExecutorServiceFuturePool.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/common/TestExecutorServiceFuturePool.java
@@ -19,18 +19,19 @@
 
 package org.apache.hadoop.fs.common;
 
-import org.apache.hadoop.test.AbstractHadoopTestBase;
-import org.apache.hadoop.test.LambdaTestUtils;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.hadoop.test.AbstractHadoopTestBase;
+import org.apache.hadoop.test.LambdaTestUtils;
 
 import static org.junit.Assert.assertTrue;
 

